src/coingecko/Coingecko.ts

```diff
@@ -113,15 +115,28 @@ export class Coingecko {
 
   async call(path: string) {
     const sendRequest = async () => {
+      const { host, proHost } = this;
+
+      // Try basic, and then pro as fallback.
+      this.logger.debug({ at: "sdk-v2/coingecko", message: `Sending GET request to host ${host}` });
```

Comment:
This can add latency. Did you do some test runs to see how much latency this would add when basic fails? You can simulate that by sending enough traffic that the basic endpoint rate limits you.

Reply:
how else would you only use pro API as a fallback without adding latency?

Reply:
I can force a 1000ms timeout on the basic request also

Reply:
You can set a relatively short timeout on the basic request to limit latency added. How important is this change though when Vercel already has caching? An alternative is to lengthen the cache TTL for the coingecko endpoint right?

Reply:
1000ms is too high IMO. That adds a whole second to latency in the worst case. We should cap in on P50 or less of the basic coingecko request.

Reply:
Actually, when empirically testing the latency isn't affected by setting a timeout on the first call. My local tests show that falling back to pro API changes latency from 150-200ms to 300-350ms

Reply:
The caching is only for 5 mins. This change would reduce the cg-pro requests a lot because most of the time the basic api works: How important is this change though when Vercel already has caching?

i can cap it at 200ms, 50% of the requests fall under this

Reply:
Is it worth it though to reduce requests to CG Pro this way? IMO, API performance seems more important than saving some amount of requests that miss the edge cache.

Reply:
If we create very low timeouts, we may _increase_ average latency because it causes us to fallback unnecessarily and force us to start a new request.

How often does this first endpoint fail? 5% of the time? 1% of the time? If it's 5% or 1% then it shouldn't have a huge impact on average latency.

Reply:
According to our usage data, the caching isn't really reducing requests as much as it should, it seems. We've used 350k requests in a 9 days, right? That is far more than 1 request per token per 5 minutes, no?

Reply:
Do we know why that is?

Reply:
![Screen Shot 2022-08-17 at 12 46 51](https://user-images.githubusercontent.com/9457025/185196656-70c25b31-d5d4-4177-b88a-2f4c77612541.png)


Quick look at datadog dashboard shows cache getting hit a lot